pkg/client/cloudevents/codec.go

```diff
@@ -0,0 +1,154 @@
+package cloudevents
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cloudeventstypes "github.com/cloudevents/sdk-go/v2/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	cegeneric "open-cluster-management.io/api/cloudevents/generic"
+	cetypes "open-cluster-management.io/api/cloudevents/generic/types"
+	workpayload "open-cluster-management.io/api/cloudevents/work/payload"
+	workv1 "open-cluster-management.io/api/work/v1"
+
+	"github.com/openshift-online/maestro/pkg/api"
+)
+
+type Codec struct{}
+
+var _ cegeneric.Codec[*api.Resource] = &Codec{}
+
+func (codec *Codec) EventDataType() cetypes.CloudEventsDataType {
+	return workpayload.ManifestEventDataType
+}
+
+func (codec *Codec) Encode(source string, eventType cetypes.CloudEventsType, obj *api.Resource) (*cloudevents.Event, error) {
+	evtBuilder := cetypes.NewEventBuilder(source, eventType).
+		WithResourceID(obj.ID).
+		WithResourceVersion(int64(obj.Version)).
+		WithClusterName(obj.ConsumerID)
```

Comment:
will consumerID be a cluster name?

Reply:
currently yes, we use consumerID as cluster name.